### ISSUE TITLE
Tender: Procurement method sub-type

### ISSUE THREAD

**Issue by yolile** [open-contracting/standard#750]
In Uruguay, apart form the procurementMethod, each of them has a sub type of procurement, that have their own codes defined here: https://www.comprasestatales.gub.uy/comprasenlinea/jboss/reporteSubTiposCompra.do

The proposal is to add a **tender/procurementSubMethod** field to add this information as an open codelist 

----

Main types are defined at https://www.comprasestatales.gub.uy/comprasenlinea/jboss/reporteTiposCompra.do

**Comment by jpmckinney** [comment]
Isn't our typical guidance to use `procurementMethodDetails` for the sub-type? I know some implementations do that.

For reference: "Additional detail on the procurement method used. This field may be used to provide the local name of the particular procurement method used."

**Comment by yolile** [comment]
No, it is not the same. They have their local procurementMethod type, using the procurementMethodDetails to publish it, but they also have another field, the subtype.

**Comment by jpmckinney** [comment]
Aha – so rather than put all the information about both the local method and submethod into one field (`procurementMethodDetails`), they prefer to maintain the structure.

This would be comparable to how Ukraine adds a field for sub-statuses: https://github.com/open-contracting/standard/issues/151#issuecomment-62179613

I think `procurementSubMethod` sounds fine. What would the codelist look like?

**Comment by yolile** [comment]
Their subtypes are defined here: https://www.comprasestatales.gub.uy/comprasenlinea/jboss/reporteSubTiposCompra.do (the id could be the code and the field "resumen" the description) but it is a long list

**Comment by jpmckinney** [comment]
Hmm, if all the codes will be local, and if the goal is to have a reusable community extension, then perhaps, at first, the code field won't have a codelist (or will have an empty codelist), until we can identify codes that are re-used across publishers. Each publisher might want to have an additional, local extension that constrains the code field using an `enum`.

I notice that some codes are repeated (e.g. id=COM, resumen= Común), which suggests that the code can only be properly interpreted in the context of the parent code – right? One option is to prefix the parent code to the code, so that the code can be interpreted in isolation. Another option is to have guidance in the publication policy about using the parent code to interpret the code.

**Comment by duncandewhurst** [comment]
I noticed that a lot of the codes seem to reference specific laws or regulations.

Have we considered whether the [covered by extension](https://github.com/open-contracting-extensions/ocds_coveredBy_extension) could be used?

Also, we already have the [procurement method modalities](https://github.com/open-contracting-extensions/ocds_procurementMethodModalities_extension/) extension which can be used to sub divide procurement methods (and has an open codelist).

For our internal reference and to help with future mapping/extension decisions, it would be good to document why a new extension is preferred over using one of the above.

**Comment by jpmckinney** [comment]
Good questions! Essentially, we need a better understanding of what the Uruguay codelist means and how it's intended to be used. Some are references to specific articles of specific laws, but others are simply "Común" ("Common"), and I don't know what that means in context.

* If the intention is to reference the law that gives the procedure authority to proceed / that the procedure must follow, then the [legalBasis](https://extensions.open-contracting.org/en/extensions/legalBasis/master/) extension is appropriate. Ideally, the code should follow a standard legal reference format (in Europe, CELEX numbers are used; if similar identifiers are available in other jurisdictions, they should be used).

* If the intention is to identify particular 'modes' of a contracting process (e.g. accelerated procedure, which can apply to many different procurement methods) and to furthermore indicate which part of the law allows that mode to be used – then it'd be better to use the more semantic [procedure](https://extensions.open-contracting.org/en/extensions/procedure/master/) extension to indicate the mode.

* If the intention is to identify the particular procedure/method used for the contracting process, then there is actually no agreed approach yet ([the EU forms](http://standard.open-contracting.org/profiles/eu/master/en/forms/F02/#IV.1.1) use `procurementMethodDetails`). This issue can be about agreeing an approach. Ideally, it would be semantic like with the [procedure](https://extensions.open-contracting.org/en/extensions/procedure/master/) extension, instead of an opaque reference to a law.

All these can be used in common, as they serve different purposes.

----

~~Note that the procurementMethodModalities extension was based primarily on GPA requirements, and hasn't yet been reviewed in light of my more recent work on EU standard forms. Some of the codes might be more appropriate to identifying a particular procedure than to describing modes that can apply to multiple procedures.~~ **Update:** https://github.com/open-contracting/standard/issues/750#issuecomment-555206636

**Comment by jpmckinney** [comment]
Sorry, I meant to refer to a proposed legalBasis extension, not coveredBy.

coveredBy was originally authored to more specifically identify processes that are "covered" by a treaty (like the GPA). The GPA doesn't give that process its legal authority, and it doesn't describe the fundamental rules of how the procedure works. That is what the legal basis does. Instead, the GPA adds additional requirements that the process must fulfill, and identifies it as an opportunity for international bidders.

I think we (myself included) might have given different guidance in the CRM on the use of coveredBy, which is a reflection of its status as a draft extension (i.e. not one that is in the registry). However, I think the distinction above is important between the legal basis (what gives authority) and any treaty coverage, as the two have rather different use cases.

**Comment by duncandewhurst** [comment]
Ah, thanks for clarification I'm not sure I was aware of legalBasis.

Is there an index somewhere of all the extensions (regardless of status) that we can refer to when considering mappings, requests for new fields etc?

I looked in https://github.com/open-contracting-extensions and couldn't find legalBasis

**Comment by jpmckinney** [comment]
legalBasis is just an issue. The only repositories with issues about proposed (non-existent) extensions are this repo and https://github.com/open-contracting-extensions/european-union/issues (to avoid adding many jurisdiction-specific issues to the standard repo).

Otherwise, all OCP-authored extensions are in https://github.com/open-contracting-extensions. CRM issue 4329 links to a spreadsheet of externally authored extensions; if an external extension is relevant to other publishers, I've updated/created issues on this repo, though I might have missed some.

In other words, you should only need to check this repo, the open-contracting-extensions organization, and (until the EU mapping is complete) the european-union repo.

**Comment by jpmckinney** [comment]
See comment in https://github.com/open-contracting/standard/issues/695#issuecomment-555205138 for newer guidance that replace the procurement method modalities proposal.

**Comment by jpmckinney** [comment]
I had another look, and as far as standardization goes, I don't think their "subtypes" would be modelled as such internationally.

There is a "Compra por Excepción" type with 82 out of the 145 subtypes. Each of those is essentially an article describing a justification for the exception. The justification for an exception isn't a "sub-type" of a procedure; it's a rationale for the procedure. For the EU, we added a [procurement method rationale classification](https://extensions.open-contracting.org/en/extensions/procurementMethodRationaleClassifications/master/) extension, so that justifications can be classified in a local standard.

"Procedimiento Especial" has 25 subtypes, which might have similar semantics as above.

That leaves 38 subtypes for the other 21 types. Some of these subtypes refer to techniques (Acuerdo Marco, Acuerdo Marco por Monto), or to competitive conditions (Internacional, Nacional), or to options (Opción Permuta).

It's certainly the case that, for any procedure in any country, we can specify in the data the legal articles that describe the procedure, whether it is accelerated, whether it uses framework agreements, whether it is open to international bids, etc. However, with OCDS, we're interested in semantics, not legal citations. So, I don't think we have a use case in this issue.

**Comment by jpmckinney** [comment]
Re-opening the issue, as there might be some method "sub-types" that can be standardized, and that *don't* overlap or mix the concepts of [procurement method rationale](https://extensions.open-contracting.org/en/extensions/procurementMethodRationaleClassifications/master/), [techniques](https://extensions.open-contracting.org/en/extensions/techniques/master/), [options](https://extensions.open-contracting.org/en/extensions/options/master/), target suppliers #1102, etc. 

[UNCITRAL glossary](http://www.uncitral.org/pdf/english/texts/procurem/ml-procurement-2011/Glossary-e.pdf) defines the terms below.

Some of these might be covered by [second stage description](https://extensions.open-contracting.org/en/extensions/secondStageDescription/master/) (e.g. `successiveReduction` for "Request for proposals with dialogue"), or might be better modelled as particular aspects of a method, rather than distinct sub-methods.

ID | Term | Definition | Reference
-- | -- | -- | --
68 | Request for proposals with consecutive negotiations | A method of procurement which main distinct feature is negotiation of financial terms of submissions after completion of the evaluation of technical, quality and performance characteristics of submissions. For the explanation of the terms “method of procurement” and “evaluation”, see ## 44 and 27 above. For the explanation of the term “submission”, see # 83 below. |  
69 | Request for proposals with dialogue | A method of procurement which main distinct feature is a dialogue with suppliers or contractors to obtain the most satisfactory solution to the procurement needs. For the explanation of the term “method of procurement”, see # 44 above. For the explanation of the term “supplier or contractor”, see # 85 below. | competitive dialogue (directive 2004/18/EC, article 1(11) (c))
70 | Request for proposals without negotiation | A method of procurement which main distinct feature is evaluation of financial terms of submissions after completion of the evaluation of technical, quality and performance characteristics of submissions whereas submissions are presented to the procuring entity in two separate sealed envelopes. For the explanation of the terms “method of procurement”, “evaluation” and “procuring entity”, see ## 44, 27 and 62 above. For the explanation of the term “submission”, see # 83 below. |  
71 | Request for quotations | A method of procurement which main distinct feature is submission of only one quotation by suppliers or contractors in response to request for quotations by the procuring entity; the quotation cannot be changed and be subject to negotiation (the method is available only for low-value simple off-the-shelf items). For the explanation of the terms “method of procurement” and “procuring entity”, see ## 44 and 62 above. For the explanation of the term “supplier or contractor”, see # 85 below. | shopping (the World Bank procurement guidelines)
72 | Request-for-proposals proceedings | Methods of procurement encompassing request for proposals without negotiation, request for proposals with dialogue and request for proposals with consecutive negotiations (see ## 70, 69 and 68 above) For the explanation of the term “method of procurement”, see # 44 above.

[EBRD glossary](https://www.ppi-ebrd-uncitral.com/index.php/en/procurement-glossary) lists:

UNCITRAL Model Law Terms | Description Notes | Others in use or similar Terms
-- | -- | --
Request for proposals with dialogue | The procedures are not identical | Competitive dialogue (EU)
Request for quotations | Relatively informal procedure with request for quotation directly addressed to limited number of suppliers | Shopping (MDBs)

**Comment by jpmckinney** [comment]
The original [procedure](https://github.com/open-contracting-extensions/ocds_procedure_extension) extension had codes for:

Code | Title | Description
-- | -- | --
negotiated | Negotiated procedure | The procurement method involves negotiation.		
optionallyNegotiated | Optionally negotiated procedure | The buyer may award contracts on the basis of the initial bids without negotiation

This was removed since, for the OCDS for EU extension, the presence of negotiations was sufficiently indicated by the procedure's name in `procurementMethodDetails`.

In this issue, we might end up adding fields to that extension to express "modalities" of procedures. For example, one field indicates whether a procedure is [accelerated](https://extensions.open-contracting.org/en/extensions/procedure/master/schema/) or not.

We should be careful to model these correctly. For example, the original procedure extension had codes for whether the procedure involved:

* joint procurement (which is better modelled as multiple parties with a buyer role)
* electronic auction (which is better modelled as a [technique](https://extensions.open-contracting.org/en/extensions/techniques/master/))
* successive stages ("The number of bids to be negotiated or solutions to be discussed are reduced in successive stages of the procedure") (which is better modelled as a description of the [second stage](https://extensions.open-contracting.org/en/extensions/secondStageDescription/master/schema/))

**Comment by jpmckinney** [comment]
> Re-opening the issue, as there might be some method "sub-types" that can be standardized, and that don't overlap or mix the concepts of [procurement method rationale](https://extensions.open-contracting.org/en/extensions/procurementMethodRationaleClassifications/master/), [techniques](https://extensions.open-contracting.org/en/extensions/techniques/master/), [options](https://extensions.open-contracting.org/en/extensions/options/master/), target suppliers https://github.com/open-contracting/standard/issues/1102, etc.

In the intervening years, we have not encountered such "sub-types" to be standardized, so closing again.
